test/python/circuit/test_controlled_gate.py

```diff
@@ -1438,7 +1438,6 @@ def test_control_zero_operand_gate(self, num_ctrl_qubits):
     @data(
         RXGate,
         RYGate,
-        RZGate,
```

Comment:
why did you need to remove this test?

Reply:
The test is asserting that a failure occurs, and for most of the gates in the list, they shouldn't really fail - we've got enough equivalence and translation rules that it should be possible to translate the gate into one we can represent better. The `add_control` logic is just not super smart about gate translation.

As a side effect of this PR, the RZ path now happens to convert things to `CPhaseGate` rather than `CRZGate` (or something like that), so the control works further than it did before.